### ISSUE TITLE
problems about train/test error and random seed

### ISSUE THREAD

**Issue by plumbum082** [zhangylch/EANN#2]
Hello Yaolong, I have some problems when using EANN to train an intermolecular potential and I would like to consult you for your advice. 
First, I found that in my training case (water clusters: dimer trimer tetramer and pentamer, randomly splited trainset:testset = 9:1), the energy error of testing set is always lower than that of the training set, and this confuses me and I am not sure when I should stop to prevent overfitting. Could you explain to me why this will happen? (I gave a picture of the error log below.) 
Second, I plan to train several parallel models and I think I may set different random seed to initialize neural network parameters, but I could not find where I could set. Could you tell me how can I set it or is there some alternative ways to train independent parallel models?
![EANN_logerr](https://github.com/zhangylch/EANN/assets/54125554/7980c107-83b5-4d8e-8cff-72dc2b48de60)

**Comment by zhangylch** [comment]
Hello, I'm very sorry for replying to you so late. Which code are you using? Still fortran code?

**Comment by plumbum082** [comment]
Never mind~ I am using the pytorch version.

---- Replied Message ----
| From | ***@***.***> |
| Date | 11/30/2023 06:31 |
| To | zhangylch/EANN ***@***.***> |
| Cc | plumbum082 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [zhangylch/EANN] problems about train/test error and random seed (Issue #2) |

Hello, I'm very sorry for replying to you so late. Which code are you using? Still fortran code?

**Comment by zhangylch** [comment]
> Never mind~ I am using the pytorch version.
> […](#)
> ---- Replied Message ---- | From | ***@***.***> | | Date | 11/30/2023 06:31 | | To | zhangylch/EANN ***@***.***> | | Cc | plumbum082 ***@***.***>, Author ***@***.***> | | Subject | Re: [zhangylch/EANN] problems about train/test error and random seed (Issue #2) | Hello, I'm very sorry for replying to you so late. Which code are you using? Still fortran code? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

If you are using the torch version, you can switch to https://github.com/zhangylch/REANN. This used repo is a temperatory repo. The reason for this strange behaviour is that the training error is not the true error, but the average of the different batches in each epoch. For convenience, the code does not recalculate the error for the training dataset.
Regarding the initialisation of the NN parameters, I think the fact that you run the code at different times causes the NN parameters to be initialised differently. If I'm not mistaken, the seed depends on the time in the torch.

**Comment by plumbum082** [comment]
Okay, thanks for your reply! By the way, if I switch to REANN, is the training error the true error or is it the same with this EANN version?

**Comment by zhangylch** [comment]
> Okay, thanks for your reply! By the way, if I switch to REANN, is the training error the true error or is it the same with this EANN version?

same with the EANN version
